pkg/portallocator/portallocator_test.go

```diff
@@ -626,10 +637,18 @@ func countTotalAllocatedPorts(pa *PortAllocator) int {
 	return count
 }
 
-func countTotalPorts(pa *PortAllocator) int {
+func countTotalPorts(pa *portAllocator) int {
 	count := 0
 	for _, node := range pa.portAllocations {
 		count += len(node)
 	}
 	return count
 }
+
+func portAllocatorImpl(t *testing.T, pai Interface) *portAllocator {
+	pa, ok := pai.(*portAllocator)
```

Comment:
```suggestion
	pa := pai.(*portAllocator)
```

Could you just do that? If the assertion fails, then a panic happens, and you get what I think is a pretty decent error message anyway? Minor nit though.

Reply:
Sounds good to me!

Reply:
Honestly that cleans up a lot. I realized along the way that I didn't need to define the `TestInterface` in the `cloudproduct` package either - I can define it in the test that's upcasting it.